Libraries/oneMKL/matrix_mul/matrix_mul_mkl.cpp

```diff
@@ -0,0 +1,190 @@
+//==============================================================
+// Copyright � 2020 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <limits>
+#include "mkl.h"
+#include "mkl_blas_sycl.hpp"
+
+using namespace std;
+using namespace cl::sycl;
+
+// Matrix size constants
+#define SIZE 4800  // Must be a multiple of 8.
+#define M SIZE / 8
+#define N SIZE / 4
+#define P SIZE / 2
+
+/**
+ * Perform the matrix multiplication on host to verify results from mkl.
+ */
+int VerifyResult(double *c_back);
+
+int main() {
+  //
+  // Initialize data for Gemm
+  //
+  // C = alpha * op(A) * op(B)  + beta * C
+  //
+  mkl::transpose transA = mkl::transpose::nontrans;
+  mkl::transpose transB = mkl::transpose::nontrans;
+
+  // matrix data sizes
+  int m = M;
+  int n = P;
+  int k = N;
+
+  // leading dimensions of data
+  int ldA = m;
+  int ldB = k;
+  int ldC = m;
+
+  // set scalar fp values
+  double alpha = 1.0;
+  double beta = 0.0;
+
+  // 1D arrays on host side
+  double *A;
+  double *B;
+  double *C;
+
+  A = new double[M * N];
```

Comment:
```suggestion
  auto A = new double[M * N];
```
Otherwise, while this usage of `new`/`delete []` is not great, at least it shows to programmers using this coding style themselves how to accelerate their code without changing their coding style... :-)

Reply:
If I use _auto A = new double[M * N]_ should I still use _delete[] A;_?

Reply:
All "new" calls must be paired with a "delete".       To do this automatically you would use a vector instead which would show a different coding approach.

Reply:
Then I'm keeping suggested `auto` and `delete[] A;`